packages/svelte/scripts/process-messages/templates/compile-warnings.js

```diff
@@ -28,13 +28,15 @@ function w(node, code, message) {
 	}
 	if (stack && stack.at(-1)?.has(code)) return;
 
-	warnings.push(
-		new InternalCompileWarning(
-			code,
-			message,
-			node && node.start !== undefined ? [node.start, node.end ?? node.start] : undefined
-		)
+	const warning = new InternalCompileWarning(
+		code,
+		message,
+		node && node.start !== undefined ? [node.start, node.end ?? node.start] : undefined
 	);
+
+	if (filter_warning && !filter_warning(warning)) return;
```

Comment:
If we make it required, we can get rid of `filter_warning &&`

```suggestion
	if (!filter_warning(warning)) return;
```